waiter/src/waiter/process_request.clj

```diff
@@ -303,13 +307,14 @@
   "Returns a channel that will contain the ByteBuffers read from the input stream.
    The input stream is read asynchronously by creating tasks on the provided executor.
    It will report any errors while reading data on the provided abort channel."
-  [^ThreadPoolExecutor executor service-id metric-group abort-ch ^InputStream input-stream]
+  [^ThreadPoolExecutor executor service-id metric-group reservation-status-promise abort-ch ^InputStream input-stream]
```

Comment:
Does `reservation-status-promise` need to be a dependency?  Could an error be signaled in another way?